pytest_reportportal/service.py

```diff
@@ -231,6 +266,79 @@ def _stop_if_necessary(self):
         except queue.Empty:
             pass
 
+
+    @staticmethod
+    def _add_item_hier_parts_dirs(item, hier_flag, dirs_level, report_parts, dirs_parts, rp_name=""):
+
+        parts_dirs = PyTestServiceClass._get_item_dirs(item)
+        dir_path = item.fspath.new(dirname="", basename="", drive="")
+        for dir_name in parts_dirs[dirs_level:]:
+            dir_path = dir_path.join(dir_name)
+            path = str(dir_path)
+
+            if hier_flag:
+                if path in dirs_parts:
```

Comment:
what is a reason for storing data to dirs_parts variable?

Reply:
We use it to store for dirs that we already met. It allow us to do not create new hierarchy dir item each time when we see directory that we already saw for previous tests.

Reply:
E.g. 
1) dir1/dir2/module1.py::TestClass1::test1
2) dir1/dir2/module1.py::TestClass1::test2
After 1) in `dirs` we will have `{'dir1': ..., 'dir1/dir2': ... }`. And we will use in 2)

Reply:
I just wandering about advantages of storing all this objects vs creating new one every time.
Is creating new object very expensive? How much data we will store if we will run really big suite (40k tests)?

Reply:
This data structure needs to correctly report hierarchy of directories. By this object we identify hierarchy item that uses in other test items. Otherwise we will report incorrectly each directory for each module/test as new hierarchy item in RP.
Check line 160: 
`self._finish_stack.extend(parts_out)`
Check line 151: 
```python
if part in self._start_stack:
    # If we've seen this part, skip it
    continue
```
Size of `dirs_parts` dict depends on tests file hierarchy. Even for 40k files it could contain only few keys (key is relative directory path).